consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +492,16 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// Note that result should NOT be nil, one edge case is when the validator re-start in the middle of a sequence.
+	// When happening, we re-verify the proposal and write it.
+	if result == nil {
```

Comment:
The comment here is confusing to me. If the result should not be nil, then we shouldn't allow it to be nil, we should panic. If however it is a valid case that the result can be nil, then the comment needs to be changed, it could simply explain the case where the result is nil.

Reply:
Addressed on https://github.com/celo-org/celo-blockchain/pull/1394/commits/e5556b16439b86a66524eb2c11c91cee8a338088